### PR TITLE
What3Words Geocoder: Add standardblend API

### LINKED ISSUE #2935: What3Words Geocoder: Fallback to StandardBlend (autosuggest) API when no results for three-word address
When a user misspells a three word address they will currently see no results. The What3Words team suggested we try faling back to their [StandardBlend](https://docs.what3words.com/api/v2/#standardblend) (Autocomplete) API in those cases to show potential matches for the given address.

If the first API call returns no results, we can make another API call and if it has results, display a Places tile view (similar to the [BikeShare Spices](https://duckduckgo.com/?q=nyc+bike+share+locations&ia=bikesharing)) which provides the similar addresses.

This will require creating another `Spice alt_to` endpoint for the StandardBlend API Endpoint.

We should use the `text` template for the tiles. The title should be the 3-Word address, and the subtitle should be the location the place is near to (provided by the API). If possible, it would be nice to use our flag icons as well, if possible but that might be difficult.

---

IA Page:  http://duck.co/ia/view/what3words 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @moollaza

### PR DESCRIPTION
**Pull request by vovkman**
If the what3words initial api call doesn't return results then a second api call is made to StandardBlend. If StandardBlend returns suggestions for three-word addresses then they are displayed in a tile view and can be selected.

Fixes #2935
Abandoned PR:  #2960

IA Page: http://duck.co/ia/view/what3words

Maintainer: @moollaza

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vovkman Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/what3words/what3words.js [hunk 1]
```diff
@@ -56,10 +66,60 @@
...
+                                displayLatLon: item.geometry.lat + '\xB0, ' + item.geometry.lng + '\xB0',
+                                place: item.place,
+                                icon: DDG.settings.region.getLargeIconURL(item.country)
+                                }
+                            }
+                        });
+                   
+                    
+                }).fail(function() {
+                    console.log("error");
```
Comment (moollaza) [inline share/spice/what3words/what3words.js]: @vovkman this should be `spice.failed('what_three_words')`

File: share/spice/what3words/what3words.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
...
         }
+        else if (api_result.status.code == 300 && api_result.status.message === "Invalid or non-existent 3 word address"){
+            // Grab 3 word address from query
+            var query = DDG.get_query()
+            console.log(query)
+            var three_word_re = /[a-z][a-z]+\.[a-z]+\.[a-z]+/;
+            var address = query.match(three_word_re)[0]
+        }
+        else {
+            var success = 'yes'
```
Comment (moollaza) [inline share/spice/what3words/what3words.js]: @vovkman this variable is only defined conditionally. Meaning "success" on line 18 has the potential to be undefined which we don't want.

You want to initialize the variable, then set it conditionally.

Comment (moollaza) [inline share/spice/what3words/what3words.js]: Also, there's no need to set it to a string, using `true` or `false` is ideal

File: share/spice/what3words/what3words.js [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
     env.ddg_spice_what3words = function (api_result) {
-
-        if (!api_result || !api_result.geometry) {
+        if (!api_result) {
             return Spice.failed('what3words');
         }
+        else if (api_result.status.code == 300 && api_result.status.message === "Invalid or non-existent 3 word address"){
+            // Grab 3 word address from query
+            var query = DDG.get_query()
+            console.log(query)
```
Comment (moollaza) [inline share/spice/what3words/what3words.js]: @vovkman please remove `console.log()`

File: share/spice/what3words/what3words.js [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
+            var three_word_re = /[a-z][a-z]+\.[a-z]+\.[a-z]+/;
+            var address = query.match(three_word_re)[0]
```
Comment (moollaza) [inline share/spice/what3words/what3words.js]: missing semi-colon: `;`

File: share/spice/what3words/what3words.js (hunk 2 above)
Comment (moollaza) [inline share/spice/what3words/what3words.js]: missing semi-colon here too

File: share/spice/what3words/what3words.js [hunk 5]
```diff
@@ -56,10 +66,60 @@
...
+                                lat: item.geometry.lat,
+                                lon: item.geometry.lng,
+                                title: item.words,
+                                description: 'near ' + item.place + ', ' + country_map[item.country],
+                                displayLatLon: item.geometry.lat + '\xB0, ' + item.geometry.lng + '\xB0',
+                                place: item.place,
+                                icon: DDG.settings.region.getLargeIconURL(item.country)
+                                }
+                            }
+                        });
```
Comment (moollaza) [inline share/spice/what3words/what3words.js]: Indentation is too deep for this and the above two lines

File: share/spice/what3words/what3words.js (hunk 1 above)
Comment (moollaza) [inline share/spice/what3words/what3words.js]: Actually, I don't think this request can fail. The data might come back as undefined though. Please confirm.

**Comment by moollaza** [comment]
This PR is being closed due to inactivity. If you're still interested in finishing, please let me know and I'll gladly re-open it for you!

**Pull request closed without merging**